linkerd/stack/src/switch_ready.rs

```diff
@@ -0,0 +1,362 @@
+use futures::future::Either;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+use tokio::time::{delay_for, Delay, Instant};
+
+#[derive(Debug, Clone)]
+pub struct SwitchReadyLayer<B> {
```

Comment:
For now, let's skip the impl of SwitchReadyLayer -- this is something we can easily do via layer::mk as needed for now, and start with a `SwitchReady::new` instead.

Reply:
yup, i considered using `layer::mk`, and can happily make that change now.